File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.HashMap;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        HashMap<Integer, Integer> map = new HashMap();
        for(int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        if(map.size() >= nums.length/2) {
            answer = nums.length/2;
        } else {
            answer = map.size();
        }
        
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    
    a = {}
    
    # 폰켓몬 종류 별 개수
    for i in range(len(nums)):
        a[nums[i]] = a.get(nums[i], 0) + 1
    
    if(len(nums)/2 < len(a)):
        answer = len(nums)/2
    else:
        answer = len(a)
        
    return answer